src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts

```diff
@@ -94,131 +112,91 @@ export class SketchStyleParserService {
-    this.parseParagraphStyle(obj, root);
-    this.parseTextFont(obj, root);
-    this.parseAttributeString(obj, root);
+  parseTextContent(curr: SketchMSLayer, root: SketchMSLayer) {
+    (root as any).content = curr.attributedString.string;
   }
 
   /**
    * Parse text font if nothing is found
    * fallback to current page font.
-   *
-   * @param obj   The current layer
-   * @param root  The root layer
    */
-  parseTextFont(obj: SketchMSLayer, root: any) {
-    const MSAttributedStringFontAttribute =
-      obj.style.textStyle.encodedAttributes.MSAttributedStringFontAttribute;
-    if (MSAttributedStringFontAttribute.hasOwnProperty('_archive')) {
-      const parsedMSAttributedStringFontAttribute = this.binaryPlistParser.parse64Content(
-        MSAttributedStringFontAttribute._archive
-      );
-      (root.style.textStyle.encodedAttributes
-        .MSAttributedStringFontAttribute as any)._transformed = parsedMSAttributedStringFontAttribute;
-    } else if (
-      MSAttributedStringFontAttribute.hasOwnProperty('_class') &&
-      MSAttributedStringFontAttribute._class === 'fontDescriptor'
-    ) {
-      this.setStyle(obj, root, {
-        'font-family': `${
-          MSAttributedStringFontAttribute.attributes.name
-        }, 'Roboto', sans-serif`,
-        'font-size': `${MSAttributedStringFontAttribute.attributes.size}px`
+  parseTextFont(curr: SketchMSLayer, root: SketchMSLayer) {
+    const obj = curr.style.textStyle.encodedAttributes.MSAttributedStringFontAttribute;
+    if (obj.hasOwnProperty('_class') && obj._class === 'fontDescriptor') {
+      this.setStyle(curr, root, {
+        'font-family': `${obj.attributes.name}, 'Roboto', sans-serif`,
+        'font-size': `${obj.attributes.size}px`
       });
+    } else if (obj.hasOwnProperty('_archive')) {
+      const archive = this.binaryPlistParser.parse64Content(obj._archive);
+      (root.style.textStyle.encodedAttributes.MSAttributedStringFontAttribute as any)._transformed = archive;
     }
   }
 
   /**
    * Parse attibutes (not used at the moment)
-   *
-   * @param obj   The current layer
-   * @param root  The root layer
    */
-  parseAttributeString(obj: SketchMSLayer, root: any) {
-    const attributedString = obj.attributedString;
-    if (attributedString.hasOwnProperty('archivedAttributedString')) {
-      const archivedAttributedString = attributedString.archivedAttributedString
-        ._archive as string;
-      const parsedArchivedAttributedString = this.binaryPlistParser.parse64Content(
-        archivedAttributedString
-      );
-      (root.attributedString
-        .archivedAttributedString as any)._transformed = parsedArchivedAttributedString;
+  parseAttributeString(curr: SketchMSLayer, root: SketchMSLayer) {
+    const obj = curr.attributedString;
+    if (obj.hasOwnProperty('archivedAttributedString')) {
+      const archive = this.binaryPlistParser.parse64Content(obj.archivedAttributedString._archive);
+      switch (archive.$key) {
```

Comment:
we should check that `archive` is truthy.

Reply:
It should be safer, i will do it.